src/fd.ts

```diff
@@ -116,7 +116,31 @@ export abstract class Fd {
   }
 }
 
+class InoIssuer {
+  // NOTE: ino 0 is reserved for the root directory
+  private next_ino: bigint = 1n;
+
+  issue(_node: unknown): bigint {
+    // TODO: consider recycling ino if the node is deallocated
+    return this.next_ino++;
+  }
+}
+
 export abstract class Inode {
+  ino: bigint;
+
+  constructor() {
+    this.ino = Inode.issue_ino(this);
+  }
+
+  static default_issuer = new InoIssuer();
```

Comment:
Maybe just have `next_ino` directly as private static field of `Inode`? `InoIssuer` seems to be an unnecessary abstraction that increases the size of the minified code.